brian2tools/mdexport/expander.py

```diff
@@ -0,0 +1,842 @@
+            if not isinstance(connector['j'], str):
+                con_str += (' to target group ' +
+                        self.check_plural(connector['j'], 'index') + ': ')
+                if hasattr(connector['j'], '__iter__'):
+                    con_str += ', '.join(
+                                    str(ind) for ind in connector['j']
+                                        )
+                else:
+                    con_str += str(connector['j'])
+            else:
+                con_str += (' to target group with generator syntax ' +
+                            connector['j'])
+
+        elif 'condition' in connector:
+            con_str += (' with condition ' +
+                        self.render_expression(connector['condition']))
+        if connector['probability'] != 1:
+            con_str += (', with probability ' +
+                        self.render_expression(connector['probability']))
+        if connector['n_connections'] != 1:
+            con_str += (', with number of connections ' +
+                        self.render_expression(connector['n_connections']))
+        if 'identifiers' in connector:
+            con_str += ('. Constants associated: ' +
+                        self.expand_identifiers(connector['identifiers']))
+        return con_str + endll
+
+    def expand_PoissonGroup(self, poisngrp):
+        """
+        Expand PoissonGroup from standard dictionary
+
+        Parameters
+        ----------
+
+        poisngrp : dict
+            Standard dictionary of PoissonGroup
+        """
+
+        md_str = ''
+        md_str += (tab + 'Name ' + bold(poisngrp['name']) + ', with \
+                population size ' + bold(poisngrp['N']) +
+                ' and rate as ' + self.render_expression(poisngrp['rates']) +
+                '.' + endll)
+        if 'identifiers' in poisngrp:
+            md_str += tab + bold('Constants:') + endll
+            md_str += self.expand_identifiers(poisngrp['identifiers'])
+        if 'run_regularly' in poisngrp:
+            md_str += tab + bold('Run regularly: ') + endll
+            for run_reg in poisngrp['run_regularly']:
+                md_str += self.expand_runregularly(run_reg)
+
+        return md_str
+
+    def expand_SpikeGeneratorGroup(self, spkgen):
+        """
+        Expand SpikeGeneratorGroup from standard dictionary
+
+        Parameters
+        ----------
+
+        spkgen : dict
+            Standard dictionary of SpikeGeneratorGroup
+        """
+        md_str = ''
+        md_str += (tab + 'Name ' + bold(spkgen['name']) +
+                ', with population size ' + bold(spkgen['N']) +
+                ', has neuron' + self.check_plural(spkgen['indices']) + ': ' +
+                ', '.join(str(i) for i in spkgen['indices']) +
+                ' that spike at times ' +
+                ', '.join(str(t) for t in spkgen['times']) +
+                ', with period ' + str(spkgen['period']) +
+                '.' + endll)
```

Comment:
A `SpikeGeneratorGroup` can potentially define many spikes, so we should use the `np.set_printoptions` trick here.

Reply:
I wrote a small function to imitate the `NumPy.set_printoptions` so that, we can use it when we loop or write in a file, but am not sure that is elegant :thinking:

Reply:
I  am on my mobile right now and can't have a look at the code, but maybe you can use https://numpy.org/doc/stable/reference/generated/numpy.array2string.html ?

Reply:
Yes sure :+1:

Reply:
Thanks, I made the changes using `array2string()`